### ISSUE TITLE
Product list page breaks when Elastic Search enabled and having 600 products per page set as default value to display on product list page.

### ISSUE THREAD

**Issue by tapantechno2007** [magento/magento2#24047]
**Preconditions**

1. Magento Version

    Magento CLI 2.3.2 & 2.3-develop

2. Nginx Version

    1.14.2

3. PHP Version

    PHP 7.2.19-0ubuntu0.18.04.1 (cli) (built: Jun  4 2019 14:48:12) ( NTS )
    Copyright (c) 1997-2018 The PHP Group
    Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies with Zend OPcache v7.2.19- 
    0ubuntu0.18.04.1, Copyright (c) 1999-2018, by Zend Technologies

4. Elasticsearch Version

    6.8.2

**Steps to reproduce**

1. Log-in to Magento Backend.
2. Navigate to Stores -> Configuration -> Catalog -> Storefront.
3. Set Configuration 'Products per Page on Grid Allowed Values' - 600, 1200, 1500
4. Set Configuration 'Products per Page on Grid Default Value' - 600

![image](https://user-images.githubusercontent.com/6151323/62615016-1eeb1a80-b92a-11e9-8a3c-3a0539c95b42.png)

5. Enable Elastic Search configuration by navigating to Stores -> Configuration -> Catalog -> Catalog Search.
6. Set Search Engine configuration to be 'Elasticsearch 6.0+'

![image](https://user-images.githubusercontent.com/6151323/62615074-362a0800-b92a-11e9-9466-e6444cd18925.png)

7. Run reindexing - bin/magento indexer:reindex
8. Now visit the front-end navigate to the category page which was created with 600 or more products mapped to it.

**Expected result**

1. The category page should list all 600 or more products mapped to it.

**Actual result**
1. It's getting failed with below exception.

2 exception(s):
Exception #0 (Zend_Db_Statement_Exception): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'FIELD(e.entity_id,14,19,22,40,41,44,48,52,84,116,132,260,340,356,372,449,450,452,458,460,468,469,470,478,485,486,487,491,499,512,520,526,531,533,535,555,559,571,572,575,582,588,591,595,610,613' in 'order clause', query was: SELECT `e`.*, `price_index`.`price`, `price_index`.`tax_class_id`, `price_index`.`final_price`, IF(price_index.tier_price IS NOT NULL, LEAST(price_index.min_price, price_index.tier_price), price_index.min_price) AS `minimal_price`, `price_index`.`min_price`, `price_index`.`max_price`, `price_index`.`tier_price`, `stock_status_index`.`stock_status` AS `is_salable` FROM `catalog_product_entity` AS `e`
 INNER JOIN `catalog_product_index_price` AS `price_index` ON price_index.entity_id = e.entity_id AND price_index.website_id = '1' AND price_index.customer_group_id = 0
INNER JOIN `cataloginventory_stock_status` AS `stock_status_index` ON e.entity_id = stock_status_index.product_id WHERE (stock_status_index.stock_status = 1) AND (e.entity_id IN (14, 19, 22, 40, 41, 44, 48, 52, 84, 116, 132, 260, 340, 356, 372, 449, 450, 452, 458, 460, 468, 469, 470, 478, 485, 486, 487, 491, 499, 512, 520, 526, 531, 533, 535, 555, 559, 571, 572, 575, 582, 588, 591, 595, 610, 613, 617, 618, 630, 635, 644, 645, 646, 651, 656, 665, 666, 672, 681, 684, 687, 688, 693, 696, 698, 701, 705, 708, 712, 713, 728, 731, 734, 741, 742, 752, 754, 755, 759, 767, 773, 778, 782, 784, 785, 786, 787, 789, 792, 803, 805, 806, 808, 827, 828, 829, 830, 836, 840, 844, 847, 860, 886, 930, 982, 995, 1052, 1054, 1055, 1061, 1064, 1069, 1073, 1075, 1082, 1084, 1087, 1089, 1099, 1102, 1110, 1115, 1123, 1124, 1128, 1131, 1136, 1145, 1146, 1147, 1156, 1159, 1160, 1161, 1164, 1167, 1169, 1172, 1174, 1185, 1210, 1212, 1219, 1230, 1231, 1246, 1249, 1251, 1254, 1255, 1262, 1267, 1274, 1275, 1283, 1305, 1307, 1309, 1313, 1314, 1315, 1316, 1320, 1323, 1325, 1326, 1327, 1330, 1333, 1343, 1345, 1349, 1350, 1351, 1353, 1359, 1361, 1364, 1369, 1371, 1379, 1381, 1385, 1388, 1392, 1397, 1400, 1403, 1409, 1414, 1419, 1422, 1430, 1432, 1435, 1444, 1448, 1450, 1451, 1455, 1462, 1463, 1478, 1479, 1480, 1484, 1494, 1498, 1501, 1502, 1503, 1509, 1514, 1517, 1523, 1526, 1531, 1535, 1536, 1542, 1544, 1550, 1559, 1561, 1563, 1564, 1566, 1570, 1579, 1588, 1589, 1590, 1591, 1592, 1594, 1606, 1607, 1612, 1614, 1620, 1633, 1635, 1645, 1647, 1648, 1664, 1678, 1687, 1692, 1696, 1699, 1703, 1719, 1724, 1725, 1728, 1734, 1735, 1737, 1740, 1743, 1754, 1756, 1760, 1762, 1769, 1781, 1784, 1789, 1796, 1802, 1810, 1814, 1829, 1838, 1840, 1843, 1844, 1850, 1852, 1859, 1866, 1868, 1874, 1876, 1877, 1878, 1879, 1890, 1893, 1898, 1899, 1900, 1908, 1915, 1917, 1921, 1926, 1927, 1930, 1932, 1934, 1949, 1950, 1955, 1959, 1961, 1972, 1987, 1990, 1993, 2000, 2006, 2013, 2022, 2024, 2027, 2028, 2039, 5, 8, 9, 10, 12, 21, 42, 100, 164, 244, 276, 324, 451, 453, 463, 471, 480, 482, 483, 489, 490, 494, 507, 508, 514, 519, 522, 523, 524, 528, 534, 539, 545, 552, 554, 558, 565, 566, 568, 574, 576, 580, 589, 596, 600, 601, 605, 611, 615, 624, 631, 633, 640, 643, 657, 659, 683, 694, 697, 702, 703, 704, 710, 722, 729, 730, 735, 738, 739, 751, 763, 764, 775, 776, 781, 800, 810, 813, 818, 819, 820, 822, 825, 826, 832, 833, 841, 845, 846, 899, 943, 1008, 1050, 1053, 1057, 1060, 1063, 1067, 1070, 1076, 1081, 1088, 1090, 1095, 1097, 1100, 1103, 1104, 1111, 1112, 1117, 1121, 1129, 1143, 1153, 1155, 1157, 1158, 1171, 1177, 1186, 1189, 1191, 1195, 1200, 1203, 1205, 1206, 1213, 1214, 1216, 1220, 1227, 1233, 1238, 1239, 1247, 1256, 1258, 1261, 1263, 1264, 1268, 1271, 1287, 1289, 1290, 1308, 1311, 1312, 1322, 1331, 1332, 1336, 1342, 1347, 1373, 1390, 1391, 1402, 1408, 1410, 1412, 1413, 1437, 1440, 1445, 1446, 1449, 1456, 1458, 1467, 1468, 1475, 1482, 1485, 1488, 1491, 1495, 1505, 1507, 1508, 1518, 1520, 1524, 1533, 1538, 1552, 1553, 1554, 1557, 1560, 1565, 1567, 1569, 1573, 1575, 1578, 1580, 1593, 1599, 1601, 1603, 1611, 1631, 1634, 1638, 1641, 1644, 1649, 1650, 1652, 1653, 1658, 1659, 1660, 1669, 1671, 1677, 1680, 1681, 1682, 1686, 1688, 1695, 1708, 1710, 1711, 1713, 1714, 1715, 1716, 1720, 1721, 1722, 1729, 1738, 1746, 1752, 1755, 1761, 1774, 1776, 1778, 1779, 1782, 1786, 1801, 1803, 1805, 1806, 1807, 1811, 1823, 1824, 1827, 1834, 1835, 1845, 1861, 1863, 1864, 1867, 1881, 1886, 1892, 1894, 1895, 1897, 1910, 1913, 1914, 1919, 1922, 1929, 1957, 1963, 1971, 1975, 1976, 1981)) ORDER BY `FIELD(e`.`entity_id,14,19,22,40,41,44,48,52,84,116,132,260,340,356,372,449,450,452,458,460,468,469,470,478,485,486,487,491,499,512,520,526,531,533,535,555,559,571,572,575,582,588,591,595,610,613,617,618,630,635,644,645,646,651,656,665,666,672,681,684,687,688,693,696,698,701,705,708,712,713,728,731,734,741,742,752,754,755,759,767,773,778,782,784,785,786,787,789,792,803,805,806,808,827,828,829,830,836,840,844,847,860,886,930,982,995,1052,1054,1055,1061,1064,1069,1073,1075,1082,1084,1087,1089,1099,1102,1110,1115,1123,1124,1128,1131,1136,1145,1146,1147,1156,1159,1160,1161,1164,1167,1169,1172,1174,1185,1210,1212,1219,1230,1231,1246,1249,1251,1254,1255,1262,1267,1274,1275,1283,1305,1307,1309,1313,1314,1315,1316,1320,1323,1325,1326,1327,1330,1333,1343,1345,1349,1350,1351,1353,1359,1361,1364,1369,1371,1379,1381,1385,1388,1392,1397,1400,1403,1409,1414,1419,1422,1430,1432,1435,1444,1448,1450,1451,1455,1462,1463,1478,1479,1480,1484,1494,1498,1501,1502,1503,1509,1514,1517,1523,1526,1531,1535,1536,1542,1544,1550,1559,1561,1563,1564,1566,1570,1579,1588,1589,1590,1591,1592,1594,1606,1607,1612,1614,1620,1633,1635,1645,1647,1648,1664,1678,1687,1692,1696,1699,1703,1719,1724,1725,1728,1734,1735,1737,1740,1743,1754,1756,1760,1762,1769,1781,1784,1789,1796,1802,1810,1814,1829,1838,1840,1843,1844,1850,1852,1859,1866,1868,1874,1876,1877,1878,1879,1890,1893,1898,1899,1900,1908,1915,1917,1921,1926,1927,1930,1932,1934,1949,1950,1955,1959,1961,1972,1987,1990,1993,2000,2006,2013,2022,2024,2027,2028,2039,5,8,9,10,12,21,42,100,164,244,276,324,451,453,463,471,480,482,483,489,490,494,507,508,514,519,522,523,524,528,534,539,545,552,554,558,565,566,568,574,576,580,589,596,600,601,605,611,615,624,631,633,640,643,657,659,683,694,697,702,703,704,710,722,729,730,735,738,739,751,763,764,775,776,781,800,810,813,818,819,820,822,825,826,832,833,841,845,846,899,943,1008,1050,1053,1057,1060,1063,1067,1070,1076,1081,1088,1090,1095,1097,1100,1103,1104,1111,1112,1117,1121,1129,1143,1153,1155,1157,1158,1171,1177,1186,1189,1191,1195,1200,1203,1205,1206,1213,1214,1216,1220,1227,1233,1238,1239,1247,1256,1258,1261,1263,1264,1268,1271,1287,1289,1290,1308,1311,1312,1322,1331,1332,1336,1342,1347,1373,1390,1391,1402,1408,1410,1412,1413,1437,1440,1445,1446,1449,1456,1458,1467,1468,1475,1482,1485,1488,1491,1495,1505,1507,1508,1518,1520,1524,1533,1538,1552,1553,1554,1557,1560,1565,1567,1569,1573,1575,1578,1580,1593,1599,1601,1603,1611,1631,1634,1638,1641,1644,1649,1650,1652,1653,1658,1659,1660,1669,1671,1677,1680,1681,1682,1686,1688,1695,1708,1710,1711,1713,1714,1715,1716,1720,1721,1722,1729,1738,1746,1752,1755,1761,1774,1776,1778,1779,1782,1786,1801,1803,1805,1806,1807,1811,1823,1824,1827,1834,1835,1845,1861,1863,1864,1867,1881,1886,1892,1894,1895,1897,1910,1913,1914,1919,1922,1929,1957,1963,1971,1975,1976,1981)` ASC

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-19076` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by 0m3r** [comment]
Currently, it's trying to do (Note the backticks):

```
... ORDER BY `FIELD(e`.`entity_id, ..., ..., ...)`
```
This should instead be:

```
... ORDER BY FIELD(`e`.`entity_id`, ..., ..., ...)
```

Fast fix

~~~diff
-$this->collection->getSelect()->order("FIELD(e.entity_id,$orderList)");
+$this->collection->getSelect()->order(
+    new \Zend_Db_Expr('FIELD(e.entity_id, ' . $orderList . ')')
+);
~~~

**Comment by 0m3r** [comment]
Fixed in 2.4-develop 

https://github.com/magento/magento2/commit/f4de663ae5bc43a2d3bb1f09ec85599a65722691
```diff
diff --git a/app/code/Magento/Elasticsearch/Model/ResourceModel/Fulltext/Collection/SearchResultApplier.php b/app/code/Magento/Elasticsearch/Model/ResourceModel/Fulltext/Collection/SearchResultApplier.php
index ad52f81bf8ed..491d505b5bd8 100644
--- a/app/code/Magento/Elasticsearch/Model/ResourceModel/Fulltext/Collection/SearchResultApplier.php
+++ b/app/code/Magento/Elasticsearch/Model/ResourceModel/Fulltext/Collection/SearchResultApplier.php
@@ -7,8 +7,8 @@
 namespace Magento\Elasticsearch\Model\ResourceModel\Fulltext\Collection;
 
 use Magento\CatalogSearch\Model\ResourceModel\Fulltext\Collection\SearchResultApplierInterface;
-use Magento\Framework\Data\Collection;
 use Magento\Framework\Api\Search\SearchResultInterface;
+use Magento\Framework\Data\Collection;
 
 /**
  * Resolve specific attributes for search criteria.
@@ -71,7 +71,7 @@ public function apply()
         $this->collection->getSelect()->where('e.entity_id IN (?)', $ids);
         $orderList = join(',', $ids);
         $this->collection->getSelect()->reset(\Magento\Framework\DB\Select::ORDER);
-        $this->collection->getSelect()->order("FIELD(e.entity_id,$orderList)");
+        $this->collection->getSelect()->order(new \Zend_Db_Expr("FIELD(e.entity_id,$orderList)"));
     }
 
     /**
```

**Comment by gerardojofre** [comment]
https://support.magento.com/hc/en-us/articles/360035142371-Catalog-pagination-doesn-t-work-when-Elasticsearch-6-x-is-used

**Comment by ihor-sviziev** [comment]
Hi @engcom-Charlie,
Could you double check if the issue still there? We have a comments that it's already fixed in 2.4-develop.
https://github.com/magento/magento2/issues/24047#issuecomment-590399350

**Comment by engcom-Charlie** [comment]
Hello @tapantechno2007.

We are not able to reproduce this issue on the latest `2.4-develop` branch by provided steps, so we have to close it.

Please feel free to comment, reopen or create new ticket according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) 
if you are still facing this issue on the latest `2.4-develop` branch.

Thank you for contribution and collaboration!
